src/python_testing/TC_CADMIN_1_22_24.py

```diff
@@ -0,0 +1,171 @@
+        try:
+            params = await self.th1.OpenCommissioningWindow(
+                nodeid=self.dut_node_id, timeout=self.max_window_duration, iteration=10000, discriminator=self.discriminator, option=1)
+            return params
+
+        except Exception as e:
+            logging.exception('Error running OpenCommissioningWindow %s', e)
+            asserts.assert_true(False, 'Failed to open commissioning window')
+
+    async def get_window_status(self) -> int:
+        AC_cluster = Clusters.AdministratorCommissioning
+        window_status = await self.read_single_attribute_check_success(dev_ctrl=self.th1, fabric_filtered=False, endpoint=0, cluster=AC_cluster, attribute=AC_cluster.Attributes.WindowStatus)
+        return window_status
+
+    def pics_TC_CADMIN_1_22(self) -> list[str]:
+        return ["CADMIN.S"]
+
+    def steps_TC_CADMIN_1_22(self) -> list[TestStep]:
+        return [
+            TestStep(1, "Commissioning, already done", is_commissioning=True),
+            TestStep(2, "TH_CR1 opens a commissioning window on DUT_CE using ECM with a value of 900 seconds",
+                     "DUT_CE opens its Commissioning window to allow a second commissioning"),
+            TestStep(3, "TH_CR1 sends an RevokeCommissioning command to the DUT"),
+            TestStep(4, "TH_CR1 reads the window status to verify the DUT_CE window is closed",
+                     "DUT_CE windows status shows the window is closed"),
+            TestStep(5, "TH_CR1 opens a commissioning window on DUT_CE using ECM with a value of 901 seconds",
+                     "DUT_CE does not open its Commissioning window to allow a second commissioning. DUT_CE shows 'Failed to open commissioning window. Global status 0x85'"),
+            TestStep(6, "TH_CR1 reads the window status to verify the DUT_CE window is closed",
+                     "DUT_CE windows status shows the window is closed"),
+        ]
+
+    @async_test_body
+    async def test_TC_CADMIN_1_22(self):
+        self.step(1)
+        self.th1 = self.default_controller
+        self.discriminator = random.randint(0, 4095)
+
+        self.step(2)
+        await self.th1.OpenCommissioningWindow(
+            nodeid=self.dut_node_id, timeout=900, iteration=10000, discriminator=self.discriminator, option=1)
+
+        self.step(3)
+        revokeCmd = Clusters.AdministratorCommissioning.Commands.RevokeCommissioning()
+        await self.th1.SendCommand(nodeid=self.dut_node_id, endpoint=0, payload=revokeCmd, timedRequestTimeoutMs=6000)
+        # The failsafe cleanup is scheduled after the command completes, so give it a bit of time to do that
+        sleep(1)
+
+        self.step(4)
+        window_status = await self.get_window_status()
+        if window_status != Clusters.AdministratorCommissioning.Enums.CommissioningWindowStatusEnum.kWindowNotOpen:
+            asserts.fail("Commissioning window is expected to be closed, but was found to be open")
```

Comment:
I think this is the right code
```suggestion
        asserts.assert_equal(window_status, Clusters.AdministratorCommissioning.Enums.CommissioningWindowStatusEnum.kWindowNotOpen, "Commissioning window is expected to be closed, but was found to be open")
```

Reply:
Thank you, this has been tested and used in the current code!